odoo/addons/base/models/ir_http.py

```diff
@@ -296,34 +296,38 @@ def _get_record_and_check(self, xmlid=None, model=None, id=None, field='datas',
             record = self.env[model].browse(int(id))
 
         # obj exists
-        if not record or not record.exists() or field not in record:
+        if not record or field not in record:
             return None, 404
 
-        if model == 'ir.attachment':
-            record_sudo = record.sudo()
-            if access_token and not consteq(record_sudo.access_token or '', access_token):
-                return None, 403
-            elif (access_token and consteq(record_sudo.access_token or '', access_token)):
-                record = record_sudo
-            elif record_sudo.public:
-                record = record_sudo
-            elif self.env.user.has_group('base.group_portal'):
-                # Check the read access on the record linked to the attachment
-                # eg: Allow to download an attachment on a task from /my/task/task_id
-                record.check('read')
-                record = record_sudo
-
-        # check read access
         try:
-            # We have prefetched some fields of record, among which the field
-            # 'write_date' used by '__last_update' below. In order to check
-            # access on record, we have to invalidate its cache first.
-            record._cache.clear()
-            record['__last_update']
-        except AccessError:
-            return None, 403
-
-        return record, 200
+            if model == 'ir.attachment':
+                record_sudo = record.sudo()
+                if access_token and not consteq(record_sudo.access_token or '', access_token):
+                    return None, 403
+                elif (access_token and consteq(record_sudo.access_token or '', access_token)):
+                    record = record_sudo
+                elif record_sudo.public:
+                    record = record_sudo
+                elif self.env.user.has_group('base.group_portal'):
+                    # Check the read access on the record linked to the attachment
+                    # eg: Allow to download an attachment on a task from /my/task/task_id
+                    record.check('read')
+                    record = record_sudo
+
+            # check read access
+            try:
+                # We have prefetched some fields of record, among which the field
+                # 'write_date' used by '__last_update' below. In order to check
+                # access on record, we have to invalidate its cache first.
+                if not record.env.su:
+                    record._cache.clear()
+                record['__last_update']
```

Comment:
Why not this?
```suggestion
                if not record.env.su:
                    record.check_access_rights('read')
                    record.check_access_rule('read')
```

Reply:
I reverted that part for now to clean the PR and have a green runbot (tests are failing for reasons not related to the diff, such as wrong data page_id on menu etc).

But this change was not working and kept raising errors. I did not yet investigate further but I will come back to you once I do to move forward on this suggested change

Reply:
Thanks for the explanation!